Install topfind into OCAML_SITELIB instead of OCAML_CORE_STDLIB.
See also: https://github.com/ocaml/opam-repository/blob/master/packages/ocamlfind/ocamlfind.1.9.5/files/0001-Fix-bug-when-installing-with-a-system-compiler.patch
diff --git a/findlib.conf.in b/findlib.conf.in
index 261d2c8..461bafc 100644
--- a/findlib.conf.in
+++ b/findlib.conf.in
@@ -1,2 +1,3 @@
 destdir="@SITELIB@"
 path="@FINDLIB_PATH@"
+ldconf="ignore"
\ No newline at end of file
diff --git a/src/findlib/Makefile b/src/findlib/Makefile
index 84514b6..12e4ef6 100644
--- a/src/findlib/Makefile
+++ b/src/findlib/Makefile
@@ -123,8 +123,7 @@ clean:
 install: all
 	$(INSTALLDIR) "$(DESTDIR)$(prefix)$(OCAML_SITELIB)/$(NAME)"
 	$(INSTALLDIR) "$(DESTDIR)$(prefix)$(OCAMLFIND_BIN)"
- test $(INSTALL_TOPFIND) -eq 0 || $(INSTALLDIR) "$(DESTDIR)$(prefix)$(OCAML_CORE_STDLIB)"
- test $(INSTALL_TOPFIND) -eq 0 || $(CP) topfind "$(DESTDIR)$(prefix)$(OCAML_CORE_STDLIB)/"
+ test $(INSTALL_TOPFIND) -eq 0 || $(CP) topfind "$(DESTDIR)$(prefix)$(OCAML_SITELIB)/"
 	files=`$(SH) $(TOP)/tools/collect_files $(TOP)/Makefile.config \
 	findlib.cmi findlib.mli findlib.cma findlib.cmxa findlib$(LIB_SUFFIX) findlib.cmxs \
 	findlib_config.cmi findlib_config.ml topfind.cmi topfind.mli \
